internal/topic/topicreaderinternal/stream_reconnector_test.go

```diff
@@ -295,6 +301,88 @@ func TestTopicReaderReconnectorStart(t *testing.T) {
 	_ = reconnector.CloseWithError(ctx, nil)
 }
 
+func TestTopicReaderReconnectorWaitInit(t *testing.T) {
+	t.Run("OK", func(t *testing.T) {
+		mc := gomock.NewController(t)
+		defer mc.Finish()
+
+		reconnector := &readerReconnector{
+			tracer: &trace.Topic{},
+		}
+		reconnector.initChannelsAndClock()
+
+		stream := NewMockbatchedStreamReader(mc)
+
+		reconnector.readerConnect = readerConnectFuncMock(readerConnectFuncAnswer{
+			callback: func(ctx context.Context) (batchedStreamReader, error) {
+				return stream, nil
+			},
+		})
+
+		reconnector.start()
+
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+		err := reconnector.WaitInit(ctx)
+		cancel()
+		require.NoError(t, err)
+
+		// one more run is needed to check idempotency
+		ctx, cancel = context.WithTimeout(context.Background(), time.Second)
+		err = reconnector.WaitInit(ctx)
+		cancel()
+		require.NoError(t, err)
+	})
+
+	t.Run("contextDeadlineInProgress", func(t *testing.T) {
+		mc := gomock.NewController(t)
+		defer mc.Finish()
+
+		reconnector := &readerReconnector{
+			tracer: &trace.Topic{},
+		}
+		reconnector.initChannelsAndClock()
+
+		stream := NewMockbatchedStreamReader(mc)
+
+		reconnector.readerConnect = readerConnectFuncMock(readerConnectFuncAnswer{
+			callback: func(ctx context.Context) (batchedStreamReader, error) {
+				return stream, nil
+			},
+		})
+
+		ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
```

Comment:
Here and in other places with use timeout for sync state
Better to wait some internal state changes, then cancel context instead of timeout.

[Example](https://github.com/ydb-platform/ydb-go-sdk/blob/d957da36f1748a81c659ee1d366f3fecdbafe78d/internal/topic/topicreaderinternal/batcher_test.go#L167-L167)


Use timeout has bad effects:
1. Slowing test suite for constant pause
2. It is difficult to debug by hands in debugger